### PR TITLE
Add NoContent factory method to Controller

### PR DESCRIPTION
**Pull request by ryanbrandenburg**
As per #3334, adding a NoContent factory method to Controller.

### CODE REVIEW

File: test/Microsoft.AspNet.Mvc.ViewFeatures.Test/ControllerTest.cs [hunk 1]
```diff
@@ -1056,6 +1056,20 @@ public void Controller_View_WithParameterViewNameAndViewModel_SetsResultViewName
         }
 
         [Fact]
+        public void Controller_NoContent()
+        {
+            // Arrange
+            var controller = new TestableController();
+
+            // Act
+            var actualContentResult = controller.NoContent();
```
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.ViewFeatures.Test/ControllerTest.cs]: I don't like this name.

File: test/Microsoft.AspNet.Mvc.ViewFeatures.Test/ControllerTest.cs [hunk 2]
```diff
@@ -1056,6 +1056,20 @@ public void Controller_View_WithParameterViewNameAndViewModel_SetsResultViewName
...
+
+            // Assert
+            Assert.IsType<NoContentResult>(actualContentResult);
```
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.ViewFeatures.Test/ControllerTest.cs]: This isn't really needed. The compiler already did this.

**Comment by rynowak** [comment]
:shipit:

File: src/Microsoft.AspNet.Mvc.ViewFeatures/Controller.cs [hunk 3]
```diff
@@ -508,6 +508,12 @@ public virtual ContentResult Content(string content, MediaTypeHeaderValue conten
             return result;
         }
 
+        [NonAction]
+        public virtual NoContentResult NoContent()
```
Comment (pranavkm) [inline src/Microsoft.AspNet.Mvc.ViewFeatures/Controller.cs]: This needs docs.

Comment (Eilon) [inline src/Microsoft.AspNet.Mvc.ViewFeatures/Controller.cs]: Yeah feel free to steal from MSDN from Web API or MVC or wherever we might have had a similar API before.

**Comment by ryanbrandenburg** [comment]
:up: :date:

**Comment by pranavkm** [comment]
:shipit:

**Pull request merged**